l10n_it_fatturapa_in/wizard/link_to_existing_invoice.py

```diff
@@ -19,20 +24,38 @@ def get_invoice_obj(self, fatturapa_attachment):
     @api.multi
     def link(self):
         self.ensure_one()
-        active_ids = self.env.context.get('active_ids')
-        if len(active_ids) != 1:
+        fatturapa_attachment_id = self.env.context.get('active_ids', [])
+        if len(fatturapa_attachment_id) != 1:
             raise UserError(_("You can select only one XML file to link."))
-        self.invoice_id.fatturapa_attachment_in_id = active_ids[0]
         # extract pdf if attached
         fatturapa_attachment_obj = self.env['fatturapa.attachment.in']
-        for fatturapa_attachment_id in active_ids:
-            fatturapa_attachment = fatturapa_attachment_obj.browse(
-                fatturapa_attachment_id)
-            fatt = self.get_invoice_obj(fatturapa_attachment)
-            for FatturaBody in fatt.FatturaElettronicaBody:
-                self.invoice_id.set_einvoice_amount(FatturaBody)
-                # 2.5
-                AttachmentsData = FatturaBody.Allegati
-                if AttachmentsData and self.invoice_id:
-                    fatturapa_attachment_obj.extract_attachments(
-                        AttachmentsData, self.invoice_id.id)
+
+        import_wiz = self.env['wizard.import.fatturapa'] \
+            .with_context(
+                active_ids=fatturapa_attachment_id,
+                linked_invoice=self.invoice_id,
+            ) \
+            .new({
+                'e_invoice_detail_level': '2',
+            })
+        fatturapa_attachment = fatturapa_attachment_obj.browse(
+            fatturapa_attachment_id)
+        fatt = self.get_invoice_obj(fatturapa_attachment)
+        einv_nbr = len(fatt.FatturaElettronicaBody)
+        if einv_nbr > 1:
+            raise UserError(
```

Comment:
Non rimuoverei la possibilità di collegare un XML a N fatture odoo

Reply:
Qui rimuovo la possibilità di collegare un XML contenente diverse fatture a una sola fattura (`self.invoice_id`) in Odoo.
Rimuovo questa possibilità perché non mi sembra abbia senso, dici che è un caso d'uso reale?

Prima di questa modifica ad esempio, i totali di ogni `FatturaBody` venivano scritti nei campi `self.invoice_id.e_invoice_amount*` e l'ultimo sovrascriveva tutti i precedenti

Reply:
> dici che è un caso d'uso reale?

È un caso d'uso che il modulo copre:

 - Creo (o il sistema genera) diversi ordini di acquisto per lo stesso fornitore in odoo
 - Dagli ordini d'acquisto genero N fatture
 - Ricevo un XML relativo a N fatture
 - Collego l'XML alle N fatture

> Prima di questa modifica ad esempio, i totali di ogni FatturaBody venivano scritti nei campi self.invoice_id.e_invoice_amount* e l'ultimo sovrascriveva tutti i precedenti

Questo è un malfunzionamento legato all'introduzione di `self.invoice_id.e_invoice_amount`. Prima di tale modifica, il wizard di "link" non coinvolgeva i dati delle singole fatture: si limitava a collegare l'XML ad una fattura odoo.

Ora, andando a scrivere dati specfici di una fattura, nel wizard di "link" bisogna permettere all'utente di specificare quale sia la fattura dell'XML che si sta collegando alla fattura odoo.

Questo in effetti doveva  essere fatto nella PR relativa a `self.invoice_id.e_invoice_amount`, ma ce lo siamo perso.
In ogni caso, andando avanti su questa strada, è da fare.

cc @robyf70

Reply:
Ho aggiornato il comportamento, ora permetto di collegare ogni fattura del file ad una diversa fattura in Odoo.